### PR TITLE
#1432 - map Deposit/Withdrawal status for bitfinex, okcoin, kraken an…

### PR DESCRIPTION
**Pull request by kisapmata**
…d btcchina. fix/adjust unit tests for this.

### CODE REVIEW

**Comment by timmolter** [comment]
@mmazi I'll leave this one for you to merge...

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java [hunk 1]
```diff
@@ -242,32 +242,48 @@ public static Type fromString(String ledgerTypeString) {
     /** The user has requested the withdrawal or deposit, or the exchange has detected an initiated deposit,
      * but the exchange still has to fully process the funding.
      * The funds are not available to the user. The funding request may possibly still be cancelled though. */
-    PROCESSING,
+    PROCESSING("WAIT CONFIFMATION","EMAIL CONFIRMATION","VERIFYING","PENDING_APPROVAL","PENDING"),
```
Comment (mmazi) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java]: "CONFI**F**MATION", are you sure? It's "Wait Confi**r**mation" in `OkCoinRecords.java`.

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java [hunk 2]
```diff
@@ -242,32 +242,48 @@ public static Type fromString(String ledgerTypeString) {
...
-    public static Status parse(String str) {
-      if (str == null) {
-        return null;
+    private String[] statusArray;
+
+    private static final Map<String, Status> fromString = new HashMap<String, Status>();
+
+    static {
+      for (final Status status : values()){
+        fromString.put(status.toString(), status);
```
Comment (mmazi) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java]: Move this line *after* the loop below to make code more resilient to future errors (the correct enum value string should overwrite any others).

**Comment by kisapmata** [comment]
thanks for spotting that. code committed now.

**Pull request merged**